### PR TITLE
feat(ironfish): Create `wallet/getAsset`

### PR DESCRIPTION
**Pull request by rohanjadvani**
## Summary

Add an RPC endpoint to fetch asset details from the wallet. We are using this to replace the call in `wallet:burn` which currently uses the chain.

## Testing Plan

Unit tests

## Documentation

Does this change require any updates to the Iron Fish Docs (ex. [the RPC API
Reference](https://ironfish.network/docs/onboarding/rpc/chain))? If yes, link a
related documentation pull request for the website.

```
[x] Yes
```

https://github.com/iron-fish/website/pull/501

## Breaking Change

Is this a breaking change? If yes, add notes below on why this is breaking and
what additional work is required, if any.

```
[ ] Yes
```

### CODE REVIEW

File: ironfish/src/rpc/routes/wallet/getAsset.ts [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+  id: string
+  metadata: string
+  name: string
+  nonce: number
+  status: string
+  // Populated for assets the account owns
+  supply: string | null
+  // Populated once the asset has been added to the main chain
+  blockHash: string | null
+  sequence: number | null
```
Comment (hughy) [inline ironfish/src/rpc/routes/wallet/getAsset.ts:30]: There are a few differences between this response and the one from `getAssets`, should we make them the same?

Comment (rohanjadvani) [reply ironfish/src/rpc/routes/wallet/getAsset.ts]: Yep, good catch. Updated

**Review by hughy — APPROVED** [review]

File: ironfish/src/rpc/routes/wallet/getAsset.ts [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+  id: string
+}
+
+export type GetWalletAssetResponse = {
+  createdTransactionHash: string
+  creator: string
+  id: string
+  metadata: string
+  name: string
+  nonce: number
```
Comment (hughy) [inline ironfish/src/rpc/routes/wallet/getAsset.ts:25]: Should we add nonce to `getAssets` in a followup?

**Review by hughy — APPROVED** [review]

File: ironfish/src/rpc/routes/wallet/getAsset.ts (hunk 2 above)
Comment (rohanjadvani) [reply ironfish/src/rpc/routes/wallet/getAsset.ts:25]: Yeah

**Pull request merged**